meson.build

```diff
@@ -366,7 +366,8 @@ possible_common_cc_flags = [
 # and -O2; this should be enough to catch most important cases without too much
 # busywork. See https://github.com/systemd/systemd/pull/19226.
 if cc.get_id() == 'gcc' and (not '02'.contains(get_option('optimization')) or
-                             cc.version().version_compare('<10'))
+                             cc.version().version_compare('<10') or
+                             run_command('sh', '-c', '[[ "$CFLAGS" =~ -O[134sg] ]]', check : false).returncode() == 0)
```

Comment:
Afaik, you should be looking at `get_option('c_args')`.

Reply:
Yeah, that should work better:

```
$ rm -fr build; CFLAGS=-O3 meson build
...
meson.build:8: WARNING: get_option(c_args): ['-O3']
meson.build:9: WARNING: get_option(optimization): 0
```